src/measure_theory/measure/haar_of_basis.lean

```diff
@@ -105,6 +105,49 @@ begin
       neg_zero, finset.univ_unique] },
 end
 
+-- TODO: move
+lemma div_le_one_of_ge {α} [linear_ordered_field α] {a b : α}
+  (h : b ≤ a) (hb : b ≤ 0) : a / b ≤ 1 :=
+by simpa only [neg_div_neg_eq] using div_le_one_of_le (neg_le_neg h) (neg_nonneg_of_nonpos hb)
+
+-- TODO: move
+lemma le_mul_of_le_one_left {α} {a b : α} [strict_ordered_ring α] (hb : b ≤ 0) (ha : a ≤ 1) :
+  b ≤ a * b :=
+by simpa only [mul_neg, neg_le_neg_iff] using mul_le_of_le_one_left (neg_nonneg_of_nonpos hb) ha
+
+/-- The axis aligned parallelepiped over `ι → ℝ` is a cuboid. -/
+lemma parallelepiped_single {ι} [decidable_eq ι] [fintype ι] (a : ι → ℝ) :
+  parallelepiped (λ i, pi.single i (a i)) = set.uIcc 0 a :=
+begin
+  ext,
+  simp_rw [set.uIcc, mem_parallelepiped_iff, set.mem_Icc, pi.le_def, ←forall_and_distrib,
+    pi.inf_apply, pi.sup_apply, ←pi.single_smul', pi.one_apply, pi.zero_apply, ←pi.smul_apply',
+    finset.univ_sum_single (_ : ι → ℝ)],
+  split,
+  { rintros ⟨t, ht, rfl⟩ i,
+    have hi := ht i,
+    simp_rw [smul_eq_mul, pi.mul_apply],
+    cases le_total (a i) 0 with hai hai,
+    { rw [sup_eq_left.mpr hai, inf_eq_right.mpr hai],
+      refine ⟨le_mul_of_le_one_left hai hi.2, mul_nonpos_of_nonneg_of_nonpos hi.1 hai⟩ },
```

Comment:
```suggestion
      exact ⟨le_mul_of_le_one_left hai hi.2, mul_nonpos_of_nonneg_of_nonpos hi.1 hai⟩ },
```
to emphasize that you're closing the proof? Same thing below.

Reply:
Whoops, leftover from extracting the missing order lemmas.